homeassistant/components/fjaraskupan/fan.py

```diff
@@ -58,7 +57,7 @@ async def async_setup_entry(
     """Set up sensors dynamically through discovery."""
 
     def _constructor(coordinator: Coordinator):
-        return [Fan(coordinator, coordinator.device, coordinator.device_info)]
+        return [Fan(coordinator, coordinator.device_info)]
```

Comment:
Would it make sense to drop the device_info parameter?

Reply:
Perhaps. I tend to like dependency injection, so that i normally avoid object.object dependency since it often ease testing.